e3db/src/main/java/com/tozny/e3db/EncryptedRecord.java

```diff
@@ -0,0 +1,171 @@
+   * should generally not be called directly; instead consider using {@link Client#encryptRecord(String, RecordData, Map, EAKInfo)}
+   * or {@link Client#encryptExisting(Record, EAKInfo)}.
+   *
+   * @param data Encrypted data. Cannot be {@code null}.
+   * @param meta Metadata associated with the record. Note that only server-specific metadata will be ignored (i.e, {@link RecordMeta#created()},
+   *             {@link RecordMeta#lastModified()}, etc.). Cannot be {@code null}.
+   * @param signature Signature of the plaintext data and metadata. Cannot be {@code null}.
+   */
+  public EncryptedRecord(Map<String, String> data, RecordMeta meta, String signature) {
+    this.data = data;
+    this.meta = meta;
+    this.signature = signature;
+  }
+
+  private static Map<String, String> getMap(JsonNode data) {
+    Map<String, String> dataMap = new HashMap<>();
+
+    Iterator<Map.Entry<String, JsonNode>> fields = data.fields();
+    while(fields.hasNext()) {
+      Map.Entry<String, JsonNode> field = fields.next();
+      dataMap.put(field.getKey(), field.getValue().asText());
+    }
+    return dataMap;
+  }
+
+  /**
+   * Encodes this instance to a string, suitable for storage. See {@link #encode(EncryptedRecord)} for details.
+   * @return {@link #encode(EncryptedRecord)}.
+   */
+  public String encode() {
+    return encode(this);
+  }
+
+  /**
+   * Encode a record such that it can be decoded by {@link #decode(String)}. Note that this does <b>not</b> encrypt the
+   * record (use {@link Client#encryptExisting(Record, EAKInfo)} or {@link Client#encryptRecord(String, RecordData, Map, EAKInfo)}
+   * first).
+   *
+   * @param record Record to serialize. Cannot be {@code null}. {@code data()} cannot
+   *               return {@code null} (though it can return an empty map). Values returned by {@code meta()} are treated as follows:
+   *
+   *               <ul>
+   *               <li>{@code recordId()}, {@code lastModified()}, {@code version()}, and {@code created()} - Not serialized.</li>
+   *               <li>{@code type()}, {@code userId()}, and {@code writerId()} - Always serialized. Cannot be {@code null}.</li>
+   *               <li>{@code plain()} and {@code signature} - Serialized only if non-{@code null}.</li>
+   *               </ul>
+   *
+   * @return Encoded representation of the record.
+   */
+  public static String encode(EncryptedRecord record) {
+    checkNotNull(record, "record");
+
+    try {
+      HashMap<String, String> meta = new HashMap<>();
+      meta.put("type", record.meta().type());
+
+      if (record.meta().plain() != null)
+        meta.put("plain", mapper.writeValueAsString(record.meta().plain()));
+
+      meta.put("userId", record.meta().userId().toString());
+      meta.put("writerId", record.meta().writerId().toString());
+
+      HashMap<String, Object> r = new HashMap<>();
+      r.put("data", record.data());
+      r.put("meta", meta);
+      r.put("rec_sig", record.signature());
```

Comment:
Looks like you're mixing camel and snake cases?

Reply:
We use snake case in all of our serialized stuff ... but often camel case in actual code. So long as we're intentional about how and where we're using different cases I think it's fine.

Reply:
I updated in a later commit in any case.